File: morphac/planners/test/planner_test.cc
#include "Eigen/Dense"
#include "gtest/gtest.h"

#include "constructs/include/control_input.h"
#include "planners/include/planner.h"

namespace {

using Eigen::VectorXd;

using morphac::constructs::ControlInput;
using morphac::planners::Planner;

// Derived Planner class for testing.
class CustomPlanner : public Planner {
 public:
  CustomPlanner(VectorXd control_input_data)
      : Planner(), control_input_data_(control_input_data) {}

  // ControlInput computation that always returns the same control input.
  ControlInput Compute() { return ControlInput(control_input_data_); }

 private:
  VectorXd control_input_data_;
};

class PlannerTest : public ::testing::Test {
 protected:
  PlannerTest() {
    // Set random seed for Eigen.
    srand(7);
  }

  void SetUp() override {}
};

TEST_F(PlannerTest, Construction) {
  VectorXd control_input_data1 = VectorXd::Random(1);
  VectorXd control_input_data2 = VectorXd::Random(6);
  CustomPlanner planner1{control_input_data1};
  CustomPlanner planner2{control_input_data2};
}

TEST_F(PlannerTest, Compute) {
  VectorXd control_input_data1 = VectorXd::Random(1);
  VectorXd control_input_data2 = VectorXd::Random(6);
  CustomPlanner planner1{control_input_data1};
  CustomPlanner planner2{control_input_data2};

  ASSERT_TRUE(planner1.Compute().get_data().isApprox(control_input_data1));
  ASSERT_TRUE(planner2.Compute().get_data().isApprox(control_input_data2));
}

}  // namespace
int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

File: morphac/planners/python/test/planner_test.py
import numpy as np
import pytest

from morphac.constructs import ControlInput
from morphac.planners import Planner


class CustomPlanner(Planner):
    def __init__(self, control_input_data):

        Planner.__init__(self)
        self.control_input_data = control_input_data

    def compute(self):

        return ControlInput(self.control_input_data)


@pytest.fixture()
def generate_planner_list():
    p1 = CustomPlanner([-1, 2, 3])
    p2 = CustomPlanner([7., 8., 9.])

    return p1, p2


def test_compute(generate_planner_list):

    p1, p2 = generate_planner_list

    assert np.allclose(p1.compute().data, [-1., 2., 3.])
    assert np.allclose(p2.compute().data, [7, 8, 9])
